speaker/voice_profile_result.go

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
+
+package speaker
+
+import (
+	"unsafe"
+
+	"github.com/Microsoft/cognitive-services-speech-sdk-go/common"
+)
+
+// #include <stdlib.h>
+// #include <stdint.h>
+// #include <speechapi_c_common.h>
+// #include <speechapi_c_result.h>
+// #include <speechapi_c_recognizer.h>
+//
+import "C"
+
+// VoiceProfileResult contains information about result from voice profile operations.
+type VoiceProfileResult struct {
+	handle C.SPXHANDLE
+
+	// ResultID specifies the result identifier.
+	ResultID string
+
+	// Reason specifies status of speech synthesis result.
+	Reason common.ResultReason
+
+	// ErrorDetails presents error details.
+	ErrorDetails string
+
+	// Collection of additional properties.
+	Properties *common.PropertyCollection
+}
+
+// Close releases the underlying resources
+func (result VoiceProfileResult) Close() {
+	result.Properties.Close()
+	C.recognizer_result_handle_release(result.handle)
+}
+
+// NewVoiceProfileResultFromHandle creates a VoiceProfileResult from a handle (for internal use)
+func NewVoiceProfileResultFromHandle (handle common.SPXHandle) (*VoiceProfileResult, error) {
```

Comment:
same

Reply:
used in VoiceProfileClient.ResetProfileAsync and DeleteProfileAsync